src/impl_methods.rs

```diff
@@ -757,11 +753,10 @@ where
         self.collapse_axis(axis, index);
         let dim = self.dim.remove_axis(axis);
         let strides = self.strides.remove_axis(axis);
-        ArrayBase {
-            ptr: self.ptr,
-            data: self.data,
-            dim,
-            strides,
+        // safe because new dimension, strides allow access to a subset of old data
+        unsafe {
+            ArrayBase::from_data_ptr(self.data, self.ptr)
```

Comment:
The `from_data_ptr` call here shouldn't be necessary.

```suggestion
            self
```

Reply:
oh right ok, remove_axis takes &self, thanks

Reply:
fixed